### ISSUE TITLE
[clang-tidy] false-positive in `bugprone-return-const-ref-from-parameter` when overload for rvalues exists

### ISSUE THREAD

**Issue by 5chmidti** [llvm/llvm-project#90274]
```c++
#include <utility>

const int &foo(const int &val) { return val; }
int &&foo(int &&val) { return std::move(val); }
```

While it is true that a `const&` parameter is returned in the first overload, the second overload handles the case that would be the problem that this check is trying to prevent/detect.

https://godbolt.org/z/4aEh9oM18
